src/main/java/rocks/inspectit/gepard/agent/transformation/DynamicTransformer.java

```diff
@@ -51,7 +52,9 @@ public DynamicType.Builder<?> transform(
       JavaModule module,
       ProtectionDomain protectionDomain) {
     InstrumentedType currentType = new InstrumentedType(typeDescription.getName(), classLoader);
-    if (resolver.shouldInstrument(typeDescription)) {
+    ClassInstrumentationConfiguration currentConfig =
+        resolver.getClassInstrumentationConfiguration(typeDescription);
+    if (currentConfig.isActive()) {
       log.debug("Adding transformation to {}", typeDescription.getName());
 
       ElementMatcher.Junction<MethodDescription> methodMatcher =
```

Comment:
and then something like 
currentConfig().scope().getMethodMatcher() would look nice